[java] Add JSpecify annotations to Color class (Issue #14291)

spencerarq opened the pull request:
### **User description**
### 🔗 Related Issues
Fixes: #14291 
(This PR addresses the ongoing effort to adopt JSpecify annotations for null-safety.)

### 💥 What does this PR do?
This PR adds JSpecify annotations to the `org.openqa.selenium.support.Color` class, as part of the effort to improve null-safety across the codebase.

### 🔧 Implementation Notes
**1. Annotation:** The `Color` class was annotated with `@NullMarked` (defaulting parameters and return types to non-null).
**2. Build Fix:** The infrastructure required an update. I added `org.jspecify:jspecify` dependency to the `support-lib` target in `BUILD.bazel` to resolve the *indirect dependency error* from Bazel's strict policy.
**3. Motivation:** This provides better static analysis support and improves Kotlin interoperability by making nullability contracts explicit for developers using the Java bindings.

### 💡 Additional Considerations
**Verification:** All related tests were executed successfully using Bazel:
`bazel test //java/test/org/openqa/selenium/support:ColorTest`
The build completed successfully and 1 test passed.

### 🔄 Types of changes
- Cleanup (formatting, renaming)
- New feature (non-breaking change which adds functionality *and tests!*)


___

### **PR Type**
Enhancement, Documentation


___

### **Description**
- Add JSpecify `@NullMarked` and `@Nullable` annotations to Color class

- Improve null-safety and Kotlin interoperability with explicit nullability contracts

- Add `org.jspecify:jspecify` dependency to support-lib BUILD target

- Mark `Converter.getColor()` method return type as nullable


___

### Diagram Walkthrough


```mermaid
flowchart LR
  A["Color class"] -->|"Add @NullMarked"| B["Class-level annotation"]
  A -->|"Add @Nullable"| C["getColor method"]
  D["BUILD.bazel"] -->|"Add dependency"| E["org.jspecify:jspecify"]
  B --> F["Explicit nullability contracts"]
  C --> F
```



<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Color.java</strong><dd><code>Add JSpecify null-safety annotations to Color class</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/src/org/openqa/selenium/support/Color.java

<ul><li>Added <code>@NullMarked</code> class-level annotation to default parameters and <br>return types to non-null<br> <li> Added <code>@Nullable</code> annotation to <code>Converter.getColor()</code> method return type<br> <li> Imported JSpecify annotation classes (<code>NullMarked</code> and <code>Nullable</code>)</ul>


</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16626/files#diff-dfb0df30134654347360c53aa1bae9b6f67247730ea72d400270f932d8464930">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>BUILD.bazel</strong><dd><code>Add JSpecify dependency to BUILD configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/src/org/openqa/selenium/support/BUILD.bazel

<ul><li>Added <code>org.jspecify:jspecify</code> artifact dependency to support-lib <br>java_export target<br> <li> Resolves indirect dependency error from Bazel's strict dependency <br>policy</ul>


</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16626/files#diff-5da6be70ade5612211a606a5e7b4627bf699105942d7f85a4cbdc8f6df474369">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

spencerarq committed:
[java] Add JSpecify annotations to Color class

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/SeleniumHQ/selenium?pullRequest=16626) <br/>All committers have signed the CLA.

asolntsev reviewed (changes requested):
@spencerarq Please fix the formatting in BUILD.bazel:

```bazel
diff --git a/java/src/org/openqa/selenium/support/BUILD.bazel b/java/src/org/openqa/selenium/support/BUILD.bazel
index ba0a0a1..ddc2313 100644
--- a/java/src/org/openqa/selenium/support/BUILD.bazel
+++ b/java/src/org/openqa/selenium/support/BUILD.bazel
@@ -35,7 +35,7 @@ java_export(
         "//java/src/org/openqa/selenium/support/ui:components",
         "//java/src/org/openqa/selenium/support/ui:elements",
         "//java/src/org/openqa/selenium/support/ui:wait",
-        artifact("org.jspecify:jspecify")
+        artifact("org.jspecify:jspecify"),
     ],
 )
```

spencerarq committed:
build(java): remove explicit version from jspecify dependency

spencerarq commented:
Applied the requested changes: added the missing comma and updated the jspecify dependency to use `artifact("org.jspecify:jspecify")` without an explicit version.

Please let me know if anything else is needed.
